pkg/fanal/analyzer/pkg/apk/apk.go

```diff
@@ -206,3 +214,27 @@ func (a alpinePkgAnalyzer) Type() analyzer.Type {
 func (a alpinePkgAnalyzer) Version() int {
 	return analyzerVersion
 }
+
+// decodeChecksumLine decodes checksum line
+func decodeChecksumLine(line string) digest.Digest {
+	if len(line) < 2 {
+		log.Logger.Debugf("Unable to decode checksum line of apk package: %s", line)
+		return ""
+	}
+	// https://wiki.alpinelinux.org/wiki/Apk_spec#Package_Checksum_Field
+	// https://stackoverflow.com/a/71712569
+	alg := digest.MD5
+	d := line[2:]
+	if strings.HasPrefix(d, "Q1") {
+		alg = digest.SHA1
+		d = d[2:] // remove `Q1` prefix
+	}
+
+	decodedDigestString, err := base64.StdEncoding.DecodeString(d)
+	if err != nil {
+		log.Logger.Debugf("unable to decode digest: %s", err)
+		return ""
+	}
+	h := hex.EncodeToString(decodedDigestString)
+	return digest.Digest(fmt.Sprintf("%s:%s", alg, h))
```

Comment:
What if adding `digest.NewDigestFromString(alg Algorithm, h string)` or something like that?

Reply:
Done - d579f5d66b7dfd86fbc936e0ed3853562ff90e20